app/src/main/java/woowacourse/payments/ui/add/AddPaymentCardStateHolder.kt

```diff
@@ -0,0 +1,63 @@
+package woowacourse.payments.ui.add
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.Saver
+import androidx.compose.runtime.saveable.rememberSaveable
+import woowacourse.payments.domain.model.CardNumber
+import woowacourse.payments.domain.model.Expiry
+import woowacourse.payments.domain.model.PaymentCard
+import woowacourse.payments.domain.model.Pin
+import woowacourse.payments.ui.model.PaymentCardUiModel
+import woowacourse.payments.ui.model.mapper.toUiModel
+
+class AddPaymentCardStateHolder private constructor(
+    private val stateHolder: MutableState<AddPaymentCardUiState>,
+) {
+    val state: AddPaymentCardUiState get() = stateHolder.value
+
+    fun onCardNumberChange(new: String) = update { it.copy(cardNumber = new) }
+
+    fun onExpiryChange(new: String) = update { it.copy(expiry = new) }
+
+    fun onOwnerChange(new: String) = update { it.copy(owner = new) }
+
+    fun onPinChange(new: String) = update { it.copy(pin = new) }
+
+    val isCardNumberValid: Boolean
+        get() = state.cardNumber.isNotEmpty() && CardNumber.from(state.cardNumber) != null
+    val isExpiryValid: Boolean
+        get() = state.expiry.isNotEmpty() && Expiry.from(state.expiry) != null
+    val isPinValid: Boolean
+        get() = state.pin.isNotEmpty() && Pin.from(state.pin) != null
+
+    val canSave: Boolean
+        get() = isCardNumberValid && isExpiryValid && isPinValid
+
+    fun buildResult(): Result<PaymentCardUiModel> =
+        PaymentCard
+            .create(state.cardNumber, state.expiry, state.owner, state.pin)
+            .map { it.toUiModel() }
```

Comment:
```suggestion
fun buildResult(): Result<PaymentCardUiModel> =
        runCatching {
            PaymentCard
                .create(state.cardNumber, state.expiry, state.owner, state.pin)
                .toUiModel()
        }
```
domain에서 Result객체를 반환해도 괜찮나요?🧐

Reply:
성공했을 때/실패했을 때를 나눠야할 것 같아서 Result를 썼었는데 지금 보니 실패했을 때를 아예 작성하지 않기도 했고, 사용하는 위치도 적절하지 않은 것 같네요 😅

[수정했습니다!](https://github.com/woowacourse/android-payments/pull/38/commits/7dbf1c0a05e6f6ffd4317d59178902168ad72d33)